src/types/database.ts

```diff
@@ -0,0 +1,13 @@
+import Database from "bun:sqlite"
+
+export type DbAction = {
+  queryMethod: QueryMethod
+  query: string
+  params?: QueryParams[]
+  errorMessage?: string
+  database: Database
+}
+
+type QueryMethod = 'run' | 'get' | 'all'
+
+type QueryParams = string | number | boolean
```

Comment:
```suggestion
import Database from "bun:sqlite"

type QueryMethod = 'run' | 'get' | 'all'

type QueryParams = string | number | boolean

export interface DbAction {
  queryMethod: QueryMethod
  query: string
  params?: QueryParams[]
  errorMessage?: string
  database: Database
}

```
For an object I normally use `interface` instead of `type`, for all cases `type` is fine. But in theory you could still use both. 

Context:
An interface is a contract to implement a shape of the data. Use the interface to make it clear that it is intended to be implemented and used as a contract about how the object will be used.
https://stackoverflow.com/a/63579488